kyuubi-rest-client/src/test/java/org/apache/kyuubi/client/BatchRestClientTest.java

```diff
@@ -0,0 +1,240 @@
+
+    assertEquals(result.getId(), expectedBatch.getId());
+    assertEquals(result.getBatchType(), expectedBatch.getBatchType());
+    assertEquals(result.getState(), expectedBatch.getState());
+  }
+
+  @Test
+  public void getBatchByIdTest() throws IOException, KyuubiException {
+    kerberizedTestHelper.login();
+
+    Batch expectedBatch = generateTestBatch();
+    Batch result = batchRestApi.getBatchById("71535");
+
+    assertEquals(result.getId(), expectedBatch.getId());
+    assertEquals(result.getBatchType(), expectedBatch.getBatchType());
+    assertEquals(result.getState(), expectedBatch.getState());
+  }
+
+  @Test
+  public void getBatchInfoListTest() throws IOException, KyuubiException {
+    kerberizedTestHelper.login();
+
+    GetBatchesResponse expectedBatchesInfo = generateTestBatchesResponse();
+    GetBatchesResponse result = batchRestApi.getBatchInfoList("spark", 0, 10);
+
+    assertEquals(expectedBatchesInfo.getBatches().size(), result.getBatches().size());
+    assertEquals(expectedBatchesInfo.getFrom(), result.getFrom());
+    assertEquals(expectedBatchesInfo.getTotal(), result.getTotal());
+  }
+
+  @Test
+  public void getOperationLogTest() throws IOException, KyuubiException {
+    kerberizedTestHelper.login();
+
+    OperationLog expectedOperationLog = generateTestOperationLog();
+    OperationLog result = batchRestApi.getOperationLog("71535", 0, 2);
+
+    assertEquals(expectedOperationLog.getRowCount(), result.getRowCount());
+  }
+
+  @Test
+  public void deleteBatchTest() throws IOException, KyuubiException {
+    kerberizedTestHelper.login();
+    batchRestApi.deleteBatch("71535", true, "b_test");
+  }
+
+  private static Batch generateTestBatch() {
+    return generateTestBatch("71535");
+  }
+
+  private static Batch generateTestBatch(String id) {
+    Map<String, String> batchInfo = new HashMap<>();
+    batchInfo.put("id", id);
+    batchInfo.put(
+        "name",
+        "org.apache.spark.deploy.SparkSubmit --conf spark.master=local "
+            + "--conf spark.kyuubi.session.engine.spark.max.lifetime=5000 "
+            + "--conf spark.kyuubi.session.engine.check.interval=1000 "
+            + "--conf spark.yarn.tags=f3dfa392-f55f-493c-bea8-559be4ba67e8 "
+            + "--conf spark.app.name=spark-batch-submission "
+            + "--class org.apache.kyuubi.engine.spark.SparkSQLEngine "
+            + "--proxy-user anonymous "
+            + "/MySpace/kyuubi-spark-sql-engine_2.12-1.6.0-SNAPSHOT.jar");
+    batchInfo.put("state", "RUNNING");
+
+    Batch batch = new Batch(id, "spark", batchInfo, "192.168.31.130:64573", "RUNNING");
+
+    return batch;
+  }
+
+  private static GetBatchesResponse generateTestBatchesResponse() {
+    Batch b1 = generateTestBatch("1");
+    Batch b2 = generateTestBatch("2");
+    List<Batch> batches = Arrays.asList(b1, b2);
+
+    return new GetBatchesResponse(0, 2, batches);
+  }
+
+  private static OperationLog generateTestOperationLog() {
```

Comment:
we can add integration test in `kyuubi-server` module

Reply:
adding test in kyuubi-server is very useful and helped me identified a lot of places for improvement..
still working in progress....